parser/Builder.cc

```diff
@@ -1829,9 +1832,9 @@ class Builder::Impl {
         return res;
     }
 
-    bool argNameCollides(std::string name) {
+    bool argNameCollides(core::NameRef name) {
         // Ignore everything beginning with underscore.
-        return (name[0] != '_');
+        return (name.shortName(gs_)[0] != '_');
```

Comment:
I think this should be fine, as the params should all be UTF8 names. If that's not true, we can go back to making `std::string` values after we find that they're going to collide.

Reply:
Heads up that I think that this case is going to cause problems for your other change:

```ruby
def foo(_x:, _x: 5)
end
```

is going to cause a crash, too.

It looks like, wildly, the Ruby VM **also** has this "ignore `_` for the sake of duplicate checking" logic, probably so that you can do things like `def foo(_, _)`.

In these cases, Sorbet is **still** going to have to mangle rename the later keyword param's name, despite not reporting an error.

Reply:
That's a great catch, thank you!